modules/core/utils/upgrade_nomad_clients.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/bash
+set -euo pipefail
+
+read -p 'AWS_PROFILE: ' awsProfile
+read -p 'NOMAD_ADDR: ' nomadAddr
+read -p 'NOMAD_TOKEN: ' nomadToken
+read -p 'ASGName: ' ASGName
+
+export AWS_PROFILE=$awsProfile
+export NOMAD_ADDR=$nomadAddr
+export NOMAD_TOKEN=$nomadToken
+
+echo 'Getting old instance ID'
+aws autoscaling describe-auto-scaling-groups \
+    --auto-scaling-group-name $ASGName \
+    | jq --raw-output '.AutoScalingGroups[0].Instances[].InstanceId' \
+    | tee instance-ids.txt
+
+echo 'Getting AutoScalingGroups max-size and desired-capacity'
+maxSize=$( aws autoscaling describe-auto-scaling-groups \
+    --auto-scaling-group-name $ASGName \
+    | jq --raw-output '.AutoScalingGroups[0].MaxSize' )
+
+desiredCapacity=$( aws autoscaling describe-auto-scaling-groups \
+    --auto-scaling-group-name $ASGName \
+    | jq --raw-output '.AutoScalingGroups[0].DesiredCapacity' )
+
+echo 'Increasing AutoScalingGroups max-size and desired-capacity by 2'
+newMaxSize=$(($maxSize*2))
+newDesiredCapacity=$(($desiredCapacity*2))
+
+echo "MaxSize = $newMaxSize"
+echo "DesiredCapacity = $newDesiredCapacity"
+
+echo 'Updating AutoScalingGroups'
+aws autoscaling update-auto-scaling-group \
+    --auto-scaling-group-name $ASGName \
+    --max-size $newMaxSize \
+    --desired-capacity $newDesiredCapacity
+
+echo 'Checking if new nodes are ready'
+nomad node status -json > nodes.json
+count=$( tr ' ' '\n' < nodes.json | grep -c ready )
+while [[ $count -lt $newDesiredCapacity ]]; do
+  echo "Waiting, currently only $count nodes are ready"
+  sleep 10 # can increase or decrease
+  nomad node status -json > nodes.json
+  count=$( tr ' ' '\n' < nodes.json | grep -c ready )
+done
+
+echo "All $count nodes are ready"
+nomad node status
+
+echo 'Getting node-ids of the old nodes'
+while read p; do
+    jq --raw-output ".[] | select (.Name == \"${p}\") | .ID" nodes.json  >> node-ids.txt
+done < instance-ids.txt
+
+echo 'Setting old instances to retire'
+while read p; do
+  nomad node eligibility -disable "${p}"
+done < node-ids.txt
+
+echo 'Draining old instances'
+while read a && read b <&3; do
+  cont=true
+  echo "Detaching instance-ids ${a}"
+  while [ $cont != false ]; do
+    errorMessage=$( aws autoscaling detach-instances --instance-ids ${p} \
+      --auto-scaling-group-name $ASGName \
+      --should-decrement-desired-capacity 2>&1 || echo $? )
+    if echo $errorMessage | grep -q 'is not part of Auto Scaling group'; then
+      cont=false
+      echo "Detaching instance-ids ${a} completed"
+      echo $errorMessage
+    fi
```

Comment:
If the command errors out due to other reasons, won't this loop continue forever?